src/main/java/io/cryostat/MainModule.java

```diff
@@ -136,35 +135,95 @@ public void println(Exception e) {
+    }
+
+    @Provides
+    @Singleton
+    @IntoSet
+    public static PluggableTypeAdapter<?> provideJmxServiceUrlAdapter(Logger logger) {
+        return new JmxServiceUrlAdapter(logger);
+    }
+
+    @Provides
+    @Singleton
+    @IntoSet
+    public static PluggableTypeAdapter<?> provideHttpMimeTypeAdapter() {
+        return new HttpMimeTypeAdapter();
+    }
+
+    @Provides
+    @Singleton
+    @IntoSet
+    public static PluggableTypeAdapter<?> provideProbeTemplateAdapter() {
+        return new ProbeTemplateTypeAdapter();
+    }
+
+    @Provides
+    @Singleton
+    @IntoSet
+    public static PluggableJsonDeserializer<?> provideRuleDeserializer() {
+        return new RuleDeserializer();
     }
 
     // public since this is useful to use directly in tests
     @Provides
     @Singleton
     public static Gson provideGson(
-            Set<PluggableTypeAdapter<?>> extraAdapters,
+            Set<PluggableTypeAdapter<?>> typeAdapters,
+            Set<PluggableJsonDeserializer<?>> deserializers,
+            Logger logger) {
+        return gsonBuilder(typeAdapters, deserializers, logger).create();
+    }
+
+    private static GsonBuilder gsonBuilder(
+            Set<PluggableTypeAdapter<?>> typeAdapters,
             Set<PluggableJsonDeserializer<?>> deserializers,
             Logger logger) {
         GsonBuilder builder =
                 new GsonBuilder()
                         .serializeNulls()
                         .disableHtmlEscaping()
-                        .registerTypeAdapter(
-                                JMXServiceURL.class, new GsonJmxServiceUrlAdapter(logger))
-                        .registerTypeAdapter(HttpMimeType.class, new HttpMimeTypeAdapter())
-                        .registerTypeHierarchyAdapter(Path.class, new PathTypeAdapter())
-                        .registerTypeAdapter(Rule.class, new RuleDeserializer())
-                        .registerTypeAdapter(ProbeTemplate.class, new ProbeTemplateTypeAdapter());
-        for (PluggableTypeAdapter<?> pta : extraAdapters) {
+                        .registerTypeHierarchyAdapter(Path.class, new PathTypeAdapter());
+        for (PluggableTypeAdapter<?> pta : typeAdapters) {
             builder = builder.registerTypeAdapter(pta.getAdaptedType(), pta);
         }
         for (PluggableJsonDeserializer<?> pjd : deserializers) {
             builder = builder.registerTypeAdapter(pjd.getAdaptedType(), pjd);
         }
-        return builder.create();
+        return builder;
+    }
+
+    @Provides
+    @Singleton
+    @Named(GSON_INTERNAL)
+    /* provides a Gson instance which includes fields with the "transient" modifier. These fields
+     * will be excluded by the main Gson instance for most purposes, like serializing structures to
+     * send as API responses. This alternate instance can be used in cases where a field should not
+     * be exposed externally in an API response but should be serialized for local storage.
```

Comment:
> This alternate instance can be used in cases where a field should not be exposed externally in an API response but should be serialized for local storage.

It's intended for use when serializing things, like Metadata that includes SecurityContexts, that are going to be stored on disk rather than sent as an API response. I think the comment is accurate...

Reply:
Oh, whoops I just read it incorrectly, my bad.